fineract-provider/src/main/java/org/apache/fineract/cob/api/LoanCOBCatchUpApiResource.java

```diff
@@ -36,12 +36,14 @@
 import org.apache.fineract.cob.data.OldestCOBProcessedLoanDTO;
 import org.apache.fineract.cob.service.LoanCOBCatchUpService;
 import org.apache.fineract.infrastructure.core.serialization.DefaultToApiJsonSerializer;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.stereotype.Component;
 
 @Path("/v1/loans")
 @Component
 @Tag(name = "Loan COB Catch Up", description = "")
 @RequiredArgsConstructor
+@ConditionalOnProperty(value = "fineract.job.loan-cob-enabled", havingValue = "true")
```

Comment:
2 things:
- if you make this condition, the exception you created will not be thrown for catch-up API calls since the API will not even be available in the application, hence you'll get a 404,rather than a 403
- I'd prefer these properties to be checkable by the compiler, so subclassing the PropertiesCondition and using that would be better.